gradle/libs.versions.toml

```diff
@@ -31,6 +32,10 @@ kotlinpoet = "com.squareup:kotlinpoet:1.16.0"
 
 ksp = { module = "com.google.devtools.ksp:symbol-processing", version.ref = "ksp" }
 ksp-api = { module = "com.google.devtools.ksp:symbol-processing-api", version.ref = "ksp" }
+ksp-cli = { module = "com.google.devtools.ksp:symbol-processing-cmdline", version.ref = "ksp" }
+ksp-aa = { module = "com.google.devtools.ksp:symbol-processing-aa", version.ref = "ksp" }
```

Comment:
`symbol-processing-aa` may not be what you need. It is a fat jar that contains unmodified classes from Kotlin compiler. It is likely to introduce conflicts in classpath at runtime.

In contrast, `symbol-processing-aa-embeddable` has the classes from Kotlin compiler and intellij **renamed**. The entry point of KSP2 is designed to avoid those renamed dependencies, so `symbol-processing-aa` shouldn't be needed even as `compileOnly`.

...unless I made some mistakes :)

`symbol-processing-cmdline` is similar that it was linked against `kotlin-compiler` rather than `kotlin-compiler-embeddable`. It is not a fat jar though. It looks like to me that maybe it is not really needed either.

Reply:
Yes for now it's just a compileOnly dependency so we can access the new KSP2 APIs, but to be honest I wasn't sure what to reach for between aa, aa-embedable, and cmdline. Some guidance would be welcome. We essentially need two things:
1. The ability to build against KSP2 APIs 
2. The ability to run KSP2 in a test

Ideally, it would be good to be able to run these as a parameterized test, but I found these artifacts to be somewhat incompatible/overlapping, so maybe the guidance above could clarify that 🤞

Reply:
I expect everything to work with just
1. `symbol-processing-aa-embeddable`
2. `symbol-processing-common-deps`
3. `symbol-processing-api`

The following shouldn't be needed. Could you try removing them and see if everything still works?
1. `symbol-processing-aa`
2. `symbol-processing-cmdline`

Reply:
I'm able to remove those dependencies, but I'm not able to get a universal build working (i.e. where I could run them both as a parameterized test.

If I put `symbol-processing-aa-embeddable` and `symbol-processing` on the same classpath, I get this at runtime when running KSP2.

```
e: java.lang.NoSuchMethodError: 'void com.google.devtools.ksp.FileToSymbolsMap.<init>(java.io.File, org.jetbrains.kotlin.com.intellij.util.io.DataExternalizer)'
	at com.google.devtools.ksp.IncrementalContext.<init>(IncrementalContext.kt:89)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension$doAnalysis$2.invoke(KotlinSymbolProcessingExtension.kt:189)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension$doAnalysis$2.invoke(KotlinSymbolProcessingExtension.kt:186)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.handleException(KotlinSymbolProcessingExtension.kt:410)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.doAnalysis(KotlinSymbolProcessingExtension.kt:186)
```

There seems to be some conflicting API usages there. Do you know if this is something reconcilable?

Reply:
Ah, you're right. Some of the shared code between KSP1 and KSP2 got renamed in one but not the other, and caused the incompatibility 🤷 

For now, I'm not going to fix it unless there is a strong use case where users need them to be in the same classpath. Please let me know if you need it.

Reply:
It would be helpful in order to test processors against both, otherwise authors will have to do build-level guards + CI matrices to do this (see the build file for this project in this PR for an example). I understand if this isn't super feasible/not worthwhile if it's temporary though!

Reply:
@ting-yuan I updated this PR with 1.0.18 here and unified the dependencies in [6280260](https://github.com/ZacSweers/kotlin-compile-testing/pull/196/commits/628026024a5e0523d52df96b53a294a34a1e638c), but I'm unfortunately seeing what appears to be huge memory consumption and OOMs in tests. Do you know what could have changed?

Reply:
1.0.18 started to index top level names in libraries / classpath. It seems that the test `findInheritedClasspathSymbols` pulls hundreds of megabytes of classes and requires a lot of memory. Increasing  heap size to 1GB makes it pass. We'll investigate and see what can be improved.

Reply:
I am running with 8GB and still getting OOMs, something definitely seems to be wrong :/

Reply:
Filed https://github.com/google/ksp/issues/1776